cpp/log/sqlite_db-inl.h

```diff
@@ -317,9 +328,27 @@ template <class Logged>
 int64_t SQLiteDB<Logged>::TreeSize() const {
   cert_trans::ScopedLatency latency(
       latency_by_op_ms.GetScopedLatency("tree_size"));
-  std::lock_guard<std::mutex> lock(lock_);
+  std::unique_lock<std::mutex> lock(lock_);
 
   CHECK_GE(tree_size_, 0);
+  sqlite::Statement statement(
```

Comment:
Yeah, but in truth, due to taking the lock (which is held during other arbitrarily long, blocking operations), this was already blocking and potentially very much *not* cheap, despite wishing it not to be. :wink: 

The previous commits remove some of the calls, or makes sure they're not on the libevent thread.

`ClusterStateController::NodeIsStale()`, for example, is called from `HttpHandler::ProxyInterceptor()`, but specifically bounced to the thread pool. `TreeSigner::SequenceNewEntries()` is called from its own thread. They both also only check it once, not in a loop or anything like that.

I think the staleness check could be improved to be cached a bit (check a raw bool that is updated regularly, say?), but I'd leave this for later, considering how `TreeSize()` was already "secretly blocking"...